### ISSUE TITLE
jsdoc linting error "JSDoc syntax error" for parameters/returns with spaces in the type

### ISSUE THREAD

**Issue by Pomax** [eslint/eslint#7194]
Using JSDocs like this will trigger a "JSDoc syntax error":

```
/**
 * Some description
 * @param {string or boolean} param1 can be either a string or boolean representing blah blah
 * @returns {undefined}
 */
function dosomething(param1) {
  if (param1) {
    ...
  }
}
```

It would be useful to have this report that spaces are not allowed in `@param`/`@return(s)`, with a separate documentation page for that error explaining that multiple types should use `{type1|type2|...}`

**Comment by platinumazure** [comment]
Unfortunately we just report the errors that [doctrine](https://github.com/eslint/doctrine) generates, so I think this might be an upstream issue. I agree that improving those error messages would be awesome, though.

**Comment by nzakas** [comment]
Yeah, that comes directly from Doctrine, which doesn't actually give anything other than a parsing error (it's just a parser, so that makes sense). I'm not sure there's anything we can do about this given that we don't get any information about why there was an error.

**Comment by kaicataldo** [comment]
@Pomax would you like to make an issue in the [Doctrine](https://github.com/eslint/doctrine) repo?

**Comment by Pomax** [comment]
Sure thing: filed https://github.com/eslint/doctrine/issues/175 as specific issue; I don't know if a separate issue for "expose doctrine errors at all" is needed, but if so I can file that too, or I can leave that up to whoever triages the doctrine issues.

**Comment by kaicataldo** [comment]
Closing this as this is really an issue with Doctrine (and a corresponding issue has been filed there).
